mollie-payments-for-woocommerce/includes/mollie/wc/plugin.php

```diff
@@ -426,22 +426,23 @@ public static function addGateways( array $gateways ) {
      */
     public static function maybeDisableApplePayGateway(array $gateways)
     {
+        $applePayGatewayClassName = 'Mollie_WC_Gateway_Applepay';
+        $applePayGatewayIndex = array_search($applePayGatewayClassName, $gateways, true);
         $postData = (string)filter_input(
             INPUT_POST,
             self::POST_DATA_KEY,
             FILTER_SANITIZE_STRING
         ) ?: '';
+        parse_str($postData, $postData);
+
+        $applePayAllowed = isset($postData[self::POST_APPLE_PAY_METHOD_ALLOWED_KEY]) && $postData[self::POST_APPLE_PAY_METHOD_ALLOWED_KEY];
 
-        if (!$postData) {
+        if (is_admin()) {
```

Comment:
Why this change from `!$postData` to `is_admin()`? and if `is_admin()` check is ok, why here and not at the very top of the method?

Reply:
Variable intializations comes before any other logic.

Reply:
Sorry @Dinamiko you're right, precondition comes first. I updated the PR